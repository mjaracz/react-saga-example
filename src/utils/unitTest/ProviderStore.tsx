import React, { FC } from 'react'
import { Provider } from 'react-redux'
import { store } from '../../redux/store'

export const ProviderStore: FC = ({ children }) => {
  return (
    <Provider store={store}>
      { children }
    </Provider>
  )
}
