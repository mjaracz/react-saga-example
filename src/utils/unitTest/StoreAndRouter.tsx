import React, { FC } from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'
import { store } from '../../redux/store'

export const StoreAndRouter: FC = ({ children }) => {
  return (
    <Provider store={store}>
      <BrowserRouter>
        { children }
      </BrowserRouter>
    </Provider>
  )
}
