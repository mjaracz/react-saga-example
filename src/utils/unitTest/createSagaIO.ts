import { stdChannel } from 'redux-saga'
import { EventEmitter } from 'events'


export const createSagaIO = (emitter: EventEmitter) => {
  const channel = stdChannel()
  emitter.on('action', channel.put)

  return {
    channel,
    dispatch: action => emitter.emit('action', action),
    getState: () => ({})
  }
}
