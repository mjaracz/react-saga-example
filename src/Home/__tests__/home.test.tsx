import React from 'react'
import { render } from '@testing-library/react'
import { Home } from '../index'
import { StoreAndRouter } from '../../utils/unitTest/StoreAndRouter'

describe('<Home />', () => {
  it('should match the snapshot', () => {
    const { baseElement } = render(
      <StoreAndRouter>
        <Home/>
      </StoreAndRouter>
    )
    expect(baseElement).toMatchSnapshot()
  })
})
