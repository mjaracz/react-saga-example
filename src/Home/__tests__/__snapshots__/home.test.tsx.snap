// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should match the snapshot 1`] = `
<body>
  <div>
    <div
      class="home"
    >
      <header
        class="home__header"
      >
        <img
          alt="logo"
          class="header__logo"
          src="logo.svg"
        />
        <p>
          Example of React with Redux-Saga
        </p>
        <a
          class="home__link"
          href="/tasks"
          rel="noopener noreferrer"
        >
          Go to Tasks Page
        </a>
      </header>
    </div>
  </div>
</body>
`;
